### DISCUSSION TITLE
Feature request: Azure OpenAI API compatibility

### DISCUSSION THREAD

**Discussion by gqoew** [open-webui/open-webui#3459] [General]
Hi !
I use an Azure OpenAI API endpoint and I can't connect it to `open-webui`
I've seen #2308, #2007, #1667, #2283, #258 discussions and I wanted to share it would be a very valuable additional feature to your project
How can we help and unblock?

**Comment by tjbck** [comment]
Azure OpenAI Plugin "Pipe function" will be provided after the 0.3.6 launch. Stay tuned!

**Reply by kevin-gbh** [reply]
Thanks @tjbck!!

**Comment by gqoew** [comment]
Great news @tjbck 

I'm not familiar yet with the repository but I'm happy to test when ready and contribute back later when I'm more comfortable with the product

Thank you!

**Comment by ill-yes** [comment]
Any updates on this?

**Reply by chemwolf6922** [reply]
Try this https://github.com/open-webui/pipelines/blob/main/examples/pipelines/providers/azure_openai_pipeline.py. I installed this pipeline, and it works great. 
For how to install pipelines: https://docs.openwebui.com/pipelines/

**Comment by hegerdes** [comment]
I was able to integrate the azure AI models via pipelines but I had to modify them.

Azure also started to offer serverless AI deployments (pay per used tokens) and these have a different URL structure. Other folks may want to use their own domain for the AI API endpoint to I would suggest changing the three parameters `AZURE_OPENAI_ENDPOINT`, `AZURE_OPENAI_DEPLOYMENT_NAME` and `AZURE_OPENAI_API_VERSION` to just `AZURE_OPENAI_ENDPOINT`, which contains the full completion domain. 

Also the authentication expects a `Bearer Token` instead of the `api-key` header. Let me know if you are okay with this change to the existing example or if I should add new one. Would happily provide a PR if interested.

**Comment by giggio** [comment]
I was able to get it to work using the https://github.com/open-webui/pipelines/blob/main/examples/pipelines/providers/azure_openai_pipeline.py file.
You need to set the environment variables.

But it does not work with DALL-E.

**Comment by dominikriedermbw** [comment]
@tjbck in our project we are working on adding native support for Azure OpenAI without pipelines. 
Do you think this could be interesting for this repo? If yes, we could aim for a PR here.

**Reply by tjbck** [reply]
Should be supported as functions already.

**Reply by iunknown80** [reply]
@tjbck I have successfully accessed Azure OpenAI through a pipeline, but this depends on another website being operational. I explored using an Azure OpenAI using functions, and it appears that a base model is necessary for the function to operate effectively. I believe that functions may not support standalone models in the same way pipelines do. Could you please confirm if my understanding is accurate?

**Comment by saquibmansuri** [comment]
For now, until the official support for Azure OpenAI get released, I've integrated like this
https://github.com/saquibmansuri/CodeHub/tree/main/DockerComposeFiles/Open-WebUI/For-AzureOpenAI

**Comment by KartDriver** [comment]
Azure seems to support the OpenAI API now.  I'm trying to set up DeepSeek.  However, I can't get it to work because the code appends the models/chat/completions in a way that isn't compatible.   Any chance you guys can add some check boxes or options so that we can change how this works and make it compatible?

**Reply by therong537** [reply]
Yeah I am also trying to connect DeepSeek to open-webui through azure's endpoint . Have you found a solution?

**Reply by arunbugkiller** [reply]
Hi guys, can you explore LiteLLM, it is very easy to set up. You can connect to the liteLLM endpoint and then use Azure AI there.
